### PR TITLE
Fixes #365: implemented `template_disk_attachment_override`

### LINKED ISSUE #365: Add the `template_disk_attachment_override` field to the `ovirt_vm` resource
## Please describe what you would like to see

Add the `template_disk_attachment_override` field to the `ovirt_vm` resource as a set with the `disk_id`, `format`, and `sparse` options.

### PR DESCRIPTION
**Pull request by janosdebugs**
## Please describe the change you are making

This PR implements #365 and adds the `template_disk_attachment_override` parameter for the `ovirt_vm` resource. It also renames the `ovirt_template_disks` data source to `ovirt_template_disk_attachments` to be more accurate. Based on #388 .

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

File: ovirt/resource_ovirt_vm.go [hunk 1]
```diff
@@ -92,6 +92,41 @@ var vmSchema = map[string]*schema.Schema{
...
+					ForceNew:         true,
+					Description:      "ID of the disk to be changed.",
+					ValidateDiagFunc: validateUUID,
+				},
+				"format": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Disk format for the override. Can be 'raw' or 'cow'.",
+					ValidateDiagFunc: validateEnum(
```
Comment (engelmi) [inline ovirt/resource_ovirt_vm.go:114]: I think we can reuse [validateFormat](https://github.com/oVirt/terraform-provider-ovirt/blob/main/ovirt/validation.go#L66-L90) here

**Review by engelmi — APPROVED** [review]

**Pull request merged**